egs/wsj/s5/steps/nnet3/lstm/make_configs.py

```diff
@@ -191,28 +196,36 @@ def ParseLstmDelayString(lstm_delay):
     prev_layer_output = nodes.AddLdaLayer(config_lines, "L0", prev_layer_output, args.config_dir + '/lda.mat')
 
     for i in range(args.num_lstm_layers):
-	if len(lstm_delay[i]) == 2: # BLSTM layer case, add both forward and backward
+        if (args.preserve_state == "true"): # add recurrent state input-nodes
+            nodes.AddRecurrentInputNodesForLstm(config_lines, "Lstm{0}".format(i+1), args.cell_dim, args.recurrent_projection_dim)
```

Comment:
Also check if the user asked for BLSTM training and state_preserve. Using state preserving training with BLSTM is not desirable.